types/angular-resource/index.d.ts

```diff
@@ -37,23 +37,21 @@ declare module 'angular' {
         // that deeply.
         ///////////////////////////////////////////////////////////////////////////
         interface IResourceService {
-            (url: string, paramDefaults?: any,
-                /** example:  {update: { method: 'PUT' }, delete: deleteDescriptor }
-                 where deleteDescriptor : IActionDescriptor */
-                actions?: IActionHash, options?: IResourceOptions): IResourceClass<IResource<any>>;
-            <T, U>(url: string, paramDefaults?: any,
-                /** example:  {update: { method: 'PUT' }, delete: deleteDescriptor }
-                 where deleteDescriptor : IActionDescriptor */
-                actions?: IActionHash, options?: IResourceOptions): U;
-            <T>(url: string, paramDefaults?: any,
-                /** example:  {update: { method: 'PUT' }, delete: deleteDescriptor }
-                 where deleteDescriptor : IActionDescriptor */
-                actions?: IActionHash, options?: IResourceOptions): IResourceClass<T>;
+            /**
+             * A factory which creates a resource object that lets you interact with RESTful server-side data sources.
+             * @param url A parameterized URL template with parameters prefixed by : as in /user/:username
+             * @param paramDefaults Default values for url parameters.
+             * @param actions example: {update: { method: 'PUT' }, delete: deleteDescriptor } where deleteDescriptor: IActionDescriptor
+             * @param options Hash with custom settings that should extend the default $resourceProvider behavior
+             */
+            (url: string, paramDefaults?: any, actions?: IActionHash, options?: IResourceOptions): IResourceClass<IResource<any>>;
+            <T>(url: string, paramDefaults?: any, actions?: IActionHash, options?: IResourceOptions): IResourceClass<T>;
+            <T, U>(url: string, paramDefaults?: any, actions?: IActionHash, options?: IResourceOptions): U;
```

Comment:
To be honest, I'm not entirely sure. I was merely just reformatting the existing types to be more readable and to clean up the Angular 1.x typings readme. I haven't used angular-resource myself much so I'm not comfortable updating the API here but it appears this is an improper use of generics since no parameters use the generic param.